pkg/controllers/status/cluster_status_controller.go

```diff
@@ -224,6 +238,32 @@ func (c *ClusterStatusController) buildInformerForCluster(cluster *v1alpha1.Clus
 	return singleClusterInformerManager, nil
 }
 
+func (c *ClusterStatusController) initLeaseController(cluster *v1alpha1.Cluster) {
+	// If lease controller has been registered, we skip this function.
+	if _, exists := c.ClusterLeaseControllers.Load(cluster.Name); exists {
+		return
+	}
+
+	// renewInterval is how often the lease renew time is updated.
+	renewInterval := time.Duration(float64(c.ClusterLeaseDuration.Nanoseconds()) * c.ClusterLeaseRenewIntervalFraction)
+
+	nodeLeaseController := lease.NewController(
+		clock.RealClock{},
+		kubeclientset.NewForConfigOrDie(c.Config),
+		cluster.Name,
+		int32(c.ClusterLeaseDuration.Seconds()),
+		nil,
+		renewInterval,
+		util.NamespaceClusterLease,
+		util.SetLeaseOwnerFunc(c.Client, cluster.Name))
+
+	c.ClusterLeaseControllers.Store(cluster.Name, nodeLeaseController)
+
+	// start syncing lease
+	// todo(garryfang): stop the controller and delete it from `ClusterLeaseControllers`
+	go nodeLeaseController.Run(wait.NeverStop)
```

Comment:
We can use the controller's stop channel, then the lease controller will be stoped follows the controller.

What we need to `TODO` is to stop the lease controller in case of `cluster` unjoin.
It's very similar to the `informerManager`, we can do it later.

Reply:
Cool, I get you.